src/main/java/io/cdap/plugin/gcp/dataplex/source/DataplexInputFormatProvider.java

```diff
@@ -0,0 +1,117 @@
+/*
+ * Copyright © 2021 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package io.cdap.plugin.gcp.dataplex.source;
+
+import com.google.cloud.dataplex.v1.StorageSystem;
+import io.cdap.cdap.api.data.batch.InputFormatProvider;
+import io.cdap.plugin.common.batch.ConfigurationUtils;
+import io.cdap.plugin.format.avro.input.CombineAvroInputFormat;
+import io.cdap.plugin.gcp.bigquery.source.PartitionedBigQueryInputFormat;
+import io.cdap.plugin.gcp.dataplex.common.util.DataplexConstants;
+import io.cdap.plugin.gcp.dataplex.common.util.DataplexUtil;
+
+import org.apache.hadoop.conf.Configuration;
+import org.apache.hadoop.mapreduce.InputFormat;
+import org.apache.hadoop.mapreduce.InputSplit;
+import org.apache.hadoop.mapreduce.JobContext;
+import org.apache.hadoop.mapreduce.RecordReader;
+import org.apache.hadoop.mapreduce.TaskAttemptContext;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+import java.util.stream.StreamSupport;
+
+/**
+ * InputFormatProvider for DataplexSource
+ */
+public class DataplexInputFormatProvider implements InputFormatProvider {
+
+  private static final Logger LOG = LoggerFactory.getLogger(DataplexBatchSource.class);
+  private static Configuration configuration;
+  protected static Map<String, String> inputFormatConfiguration;
+
+  /**
+   * @param conf configuration
```

Comment:
nit: add small description explaining how the configuration is used to initialize this provider

Reply:
added.